lib/acme/certificate.rb

```diff
@@ -9,6 +9,39 @@ def initialize(certificate, chain, request)
     @x509 = certificate
     @x509_chain = chain
     @request = request
+
+    #create_self_signed_cert if x509.nil?
+  end
+
+  def create_self_signed_cert subjectAltDomains
+    keypair = OpenSSL::PKey::RSA.new(2048) 
+
+    cert = OpenSSL::X509::Certificate.new
+    cert.subject = cert.issuer = OpenSSL::X509::Name.parse "/CN=#{subjectAltDomains.first}"
+    cert.not_before = Time.now
+    cert.not_after = Time.now + 48*3600
```

Comment:
Two days is impractical for a general purpose API as it is now, but probably actually a bit long for a special purpose API when moved into the challenge class. It would be good to find out for how long boulder keeps a pending challenge alive.